packages/ui/src/location.ts

```diff
@@ -170,21 +170,27 @@ export function setLocationStorageKey (storageKey: string): void {
   locationStorageKeyId = storageKey
 }
 
-export function navigate (location: PlatformLocation, store = true): boolean {
+export function navigate (location: PlatformLocation, replace = false): boolean {
   closePopup()
   const cur = locationToUrl(getCurrentLocation())
   const url = locationToUrl(location)
   if (cur !== url) {
-    if (store) {
-      if (!embeddedPlatform) {
+    if (!embeddedPlatform) {
+      if (replace) {
+        history.replaceState(null, '', url)
+      } else {
         history.pushState(null, '', url)
+      }
+    } else {
+      if (replace) {
+        history.replaceState({ location }, '')
       } else {
         history.pushState({ location }, '')
       }
```

Comment:
May be something like? will be more clear
```suggestion
    const data = !embeddedPlatform ? null : { location }
    const _url = !embeddedPlatform ? url : undefined
    if (replace) {
      history.replaceState(data, '', _url)
    } else {
      history.pushState(data, '', _url)
    }
```
?